alica_engine/src/engine/Logger.cpp

```diff
@@ -26,8 +26,14 @@ namespace alica
 {
 using std::to_string;
 
-Logger::Logger(AlicaEngine* ae)
-        : _ae(ae)
+Logger::Logger(ConfigChangeListener& configChangeListener, const TeamManager& teamManager, const TeamObserver& teamObserver,
+        const PlanRepository& planRepository, const AlicaClock& clock, const std::string& localAgentName)
+        : _configChangeListener(configChangeListener)
```

Comment:
We only use the configChangeListener in the constructor. We can use the passed instance to call subscribe and reload once and get rid of storing the configChangeListener locally.